homeassistant/components/roborock/select.py

```diff
@@ -112,3 +117,40 @@ async def async_select_option(self, option: str) -> None:
     def current_option(self) -> str | None:
         """Get the current status of the select entity from device_status."""
         return self.entity_description.value_fn(self._device_status)
+
+
+class RoborockCurrentMapSelectEntity(RoborockCoordinatedEntityV1, SelectEntity):
+    """A class to let you set the selected map on Roborock vacuum."""
+
+    _attr_entity_category = EntityCategory.DIAGNOSTIC
+    _attr_translation_key = "selected_map"
+
+    def __init__(
+        self,
+        unique_id: str,
+        coordinator: RoborockDataUpdateCoordinator,
+    ) -> None:
+        """Create a select entity."""
+        super().__init__(unique_id, coordinator)
+
+    async def async_select_option(self, option: str) -> None:
+        """Set the option."""
+        for map_id in self.coordinator.maps:
+            if self.coordinator.maps[map_id].name == option:
+                await self.send(
+                    RoborockCommand.LOAD_MULTI_MAP,
+                    [map_id],
+                )
+                break
```

Comment:
```suggestion
        for map_id, map_ in self.coordinator.maps.items():
            if map_.name == option:
                await self.send(
                    RoborockCommand.LOAD_MULTI_MAP,
                    [map_id],
                )
                break
```

Use `.items()` to avoid looking up the key every loop